google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupAdvancedRemarketing.java

```diff
@@ -0,0 +1,225 @@
+                    .setOperator(UserListDateRuleItemOperator.AFTER)
+                    .setValue(StringValue.of("20191031"))
+                    .build())
+            .build();
+
+    // Creates the RuleItem for checkout end date.
+    UserListRuleItemInfo endDateRule =
+        UserListRuleItemInfo.newBuilder()
+            // The rule variable name must match a corresponding key name fired from a pixel.
+            .setName(StringValue.of("checkoutdate"))
+            .setDateRuleItem(
+                UserListDateRuleItemInfo.newBuilder()
+                    .setOperator(UserListDateRuleItemOperator.BEFORE)
+                    .setValue(StringValue.of("20190101"))
+                    .build())
+            .build();
+
+    // Creates a rule group targeting users who checked out between November and December by using
+    // the start and end date rules. Combining the two rule items into a UserListRuleItemGroupInfo
+    // object causes Google Ads to AND their rules together. To instead OR the rules together, each
+    // rule should be placed in its own rule item group.
+    UserListRuleItemGroupInfo checkoutDateRuleGroup =
+        UserListRuleItemGroupInfo.newBuilder()
+            .addAllRuleItems(ImmutableList.of(startDateRule, endDateRule))
+            .build();
+
+    // Creates an ExpressionRuleUserListInfo object, or a boolean rule that defines this user list.
+    // The default rule_type for a UserListRuleInfo object is OR of ANDs (disjunctive normal form).
+    // That is, rule items will be ANDed together within rule item groups and the groups themselves
+    // will be ORed together.
+    ExpressionRuleUserListInfo expressionRuleUserListInfo =
+        ExpressionRuleUserListInfo.newBuilder()
+            .setRule(
+                UserListRuleInfo.newBuilder()
+                    .addAllRuleItemGroups(
+                        ImmutableList.of(checkoutAndCartSizeRuleGroup, checkoutDateRuleGroup)))
+            .build();
+
+    // Defines a representation of a user list that is generated by a rule.
+    RuleBasedUserListInfo ruleBasedUserListInfo =
+        RuleBasedUserListInfo.newBuilder()
+            // Optional: To include past users in the user list, set the prepopulation_status to
+            // REQUESTED.
+            .setPrepopulationStatus(UserListPrepopulationStatus.REQUESTED)
+            .setExpressionRuleUserList(expressionRuleUserListInfo)
+            .build();
+
+    // Creates a user list.
+    UserList userList =
+        UserList.newBuilder()
+            .setName(StringValue.of("My expression rule user list #" + System.currentTimeMillis()))
+            .setDescription(
+                StringValue.of(
+                    "Users who checked out in November or December OR visited the checkout page"
+                        + " with more than one item in their cart"))
+            .setMembershipStatus(UserListMembershipStatus.OPEN)
+            .setMembershipLifeSpan(Int64Value.of(365))
```

Comment:
Why use such a large membership span for just 2 months of tracking?

Reply:
Updated to 90 days.